locked-asset/simple-lock-energy/src/penalty.rs

```diff
@@ -0,0 +1,83 @@
+use common_structs::Epoch;
+
+use crate::lock_options::EPOCHS_PER_YEAR;
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+#[derive(TypeAbi, TopEncode, TopDecode)]
+pub struct PenaltyPercentage {
+    pub first_threshold: u64,
+    pub second_threshold: u64,
+    pub third_threshold: u64,
+}
+
+#[elrond_wasm::module]
+pub trait LocalPenaltyModule {
+    fn calculate_penalty_percentage_full_unlock(
+        &self,
+        lock_epochs_remaining: Epoch,
+        penalty_percentage_struct: &PenaltyPercentage,
+    ) -> u64 {
+        let first_threshold_penalty = penalty_percentage_struct.first_threshold;
+        let second_threshold_penalty = penalty_percentage_struct.second_threshold;
+        let third_threshold_penalty = penalty_percentage_struct.third_threshold;
+
+        match lock_epochs_remaining / (EPOCHS_PER_YEAR + 1u64) {
+            0 => first_threshold_penalty * lock_epochs_remaining / EPOCHS_PER_YEAR,
+            1 => {
+                // value between 0 and 360
```

Comment:
I would remove these comments, as the formula is relatively easy to read and they do not help the reader to understand the logic more easily.